client/components/singleProduct.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import {connect} from 'react-redux'
+import {getOneProductThunk} from '../store/allProducts'
+import {Link} from 'react-router-dom'
+
+class SingleProduct extends React.Component {
+  componentDidMount() {
+    const productId = this.props.match.params.productId
+    this.props.loadOneDuck(productId)
+  }
+
```

Comment:
Could you walk us through this?

Reply:
so here when the component mounts it will take the number at the end of the url using the built in match.params and assign it to the var productId. Then the productId will be taken as an argument into the loadOneDuck function which was made in mapStatetoDispatch at the bottom of the page. The loadOneDuck function will dispatch the getOneDuck thunk creator taking productId as an argument. Then when the getOneDuck thunk creator makes the axios request, it is able to specify that it want the data for the specific productId that was passed through

Reply:
Ahhh, sorry. We meant the componentDidUpdate method! Why is that needed?

Reply:
no worries and come to think of it I don't think it's needed. I thought i needed it to change between ducks without needing a page refresh but I'm actually not having that issue with it commented out